.travis.yml

```diff
@@ -30,6 +30,8 @@ env:
   - AWS_ACCESS_KEY_ID__TO_BE_REEXPORTED_ON_DEPLOYS=AKIAV6A6G7RQ2HFZ5KP7
   - secure: oEmZoB4oP4ygCMRZp86AhB40ppH87pduS0p3zVVAnTLFLkrnA4qP0TvEOqwzc6DFceQuJ6telJOGUnB4ouFIl8aBoRW7KaIuudjVWxcBLEUDdoXlA9hGSY+BOTiBmMVX5g7Wdhfngy4nygrk01cG2UWvfm62VrexeV+48twaBJE=
   - RUST_BACKTRACE="all"
+  # TODO(tonic): Temporary measure until figure out async's that need to be Box'ed.
+  - RUST_MIN_STACK="8388608"
```

Comment:
This file isn't supposed to be edited directly: see the header.

But also, I think this is a blocker for landing: we don't control the stack size of users in the wild, and so users would need to set this environment variable in `pants` scripts, etc in order to avoid explosions. I'd also be worried about efficiency.

https://without.boats/blog/futures-and-segmented-stacks/#segmenting-your-futures-for-fun-and-profit explains the problem, and mentions that the solution is to use `BoxFuture`. It's not clear where in the stack we should do this, but it amounts to:
```rust
// from
async fn ... () -> Result<..> {
 ..
}

// to
fn ..() -> BoxFuture<Result<..>> {
  async move { .. }.boxed()
}
```

Experimenting with applying that transform in a few places and seeing how much it affects the stack size might be good?